packages/x-charts-pro/src/Heatmap/Heatmap.tsx

```diff
@@ -129,7 +130,14 @@ const Heatmap = React.forwardRef(function Heatmap(
   const clipPathId = `${id}-clip-path`;
 
   const defaultizedXAxis = React.useMemo(
-    () => xAxis.map((axis) => ({ scaleType: 'band' as const, categoryGapRatio: 0, ...axis })),
+    () =>
+      (xAxis && xAxis.length > 0 ? xAxis : [{}]).map((axis) => ({
+        id: DEFAULT_X_AXIS_KEY,
+        scaleType: 'band' as const,
+        categoryGapRatio: 0,
+        data: [],
```

Comment:
I'm surprised you had to modify the x-axis and not the y-axis since they are used the same way.

A slight improvement could be to set some default data such that he series can be display.

```suggestion
        data: Array.from(
      { length: Math.max(...series[0].data.map(([xIndex, ]) => xIndex) + 1 },
      (_, index) => index,
    ),
```

similarly for the y axis
```
        data: Array.from(
      { length: Math.max(...series[0].data.map(([_, yIndex, ]) => yIndex) + 1 },
      (_, index) => index,
    ),
```

Another improvement coudl be to add in he HeatMapPlot a warning to indicate that some indexes of the series are shight than the length of x/y axes' data. SO they can not be displayed. Otherwise data are missing without giving a clue to the developer about why

https://github.com/user-attachments/assets/ce7c638d-f8ea-4c14-9c98-6c79dd433226

Reply:
> I can't seem to find a selector for the X/Y axes without relying on CSS classes.

I don't see issues about using CSS classes. We need to keep them stable since devs are using them for customization. So using them in selectors should be fine, and has the adventage of also testing the extense of those classes :)

Reply:
@alexfauquette added the defaulting as you suggested, but changed it slightly. It'll only show values that exist in the dataset, rather than values from 0 to the greatest value in the dataset.

The problem with the latter approach is that negative numbers would be hidden.